rq/worker_registration.py

```diff
@@ -92,10 +92,22 @@ def clean_worker_registry(queue: 'Queue'):
         results = pipeline.execute()
 
         invalid_keys = []
+        keys_to_check = []
 
         for i, key_exists in enumerate(results):
             if not key_exists:
                 invalid_keys.append(keys[i])
+            else:
+                keys_to_check.append(keys[i])
+
+        if keys_to_check:
+            for to_check_subset in split_list(keys_to_check, MAX_KEYS):
```

Comment:
Do we have an idea about the time complexity of this? How often this method will be called?

Reply:
it's part of the maintenance tasks, it's run every maintenance_interval, defaulting to `DEFAULT_MAINTENANCE_TASK_INTERVAL = 10 * 60`

Reply:
the maintenance task already runs in `O(log(N)+M)` where N is the number of jobs in registries, M the number of jobs cleanups (times the number of job registries, but in theory irrelevant to compute run complexity)

the cleanup_workers_registry already runs in O(W) where W is the number of workers. This PR doesn't change the run complexity.

cleanup_workers_registry is part of the maintenance task, which has a running complexity of around:

```
O(
Q # constant, number of queues
*
(
      log(N)+M      # finished jobs registry
    + 2*log(N)+3*M  # started jobs registry
    + log(N)+M      # failed jobs registry
    + log(W)   # cleanup of workers
)
```

roughly equal to `O( Q*4*log(N)+5M+log(W) )`

where:
- N the number of jobs in a registry,
- M the number of jobs cleaned up in a registry
- W the workers count